packages/offix-datastore/src/DataStore.ts

```diff
@@ -0,0 +1,39 @@
+import { buildSchema } from "graphql";
+import { readFileSync } from "fs";
+import { createDefaultStorage, Storage } from "./storage";
+import { extractModelsFromSchema, Model, PersistedModel } from "./models";
+import { createPredicate } from "./predicates";
+
+let storage: Storage;
+
+export function configure(schemaLocation: string, schemaVersion: number = 1) {
+    const schemaText = readFileSync(schemaLocation, "utf8");
```

Comment:
I love that this is schema aware. 
So all model information will be based on the schema.

Would this happen on runtime or you see us having CLI that will store some extra metadata for models?

Reply:
I'm not sure about how much information from the schema we will need yet, right now, I only extract model types and create object stores.
Right now, I'm leaning generating types for models with codegen cli. But for most of the operations so far only the model name is required `__typename` which is used to determine the store so we might just use model names in the API. What I have here is a rough sketch to that I hope we can work on.

Reply:
From my side this is perfect. You will need to understand model names and their relationships along with the ID's for effective predicates. 

All off that can be easily done after so no worries.